### PR TITLE
client: return OC_CONNECTION_CLOSED to client handler

### PR DESCRIPTION
**Pull request by jkralik**
when connection has been closed by peer or client.

### CODE REVIEW

File: api/oc_ri.c [hunk 1]
```diff
@@ -1680,7 +1680,7 @@ oc_ri_free_client_cbs_by_mid(uint16_t mid)
     if (!cb->multicast && !cb->discovery && cb->ref_count == 0 &&
         cb->mid == mid) {
       cb->ref_count = 1;
-      notify_client_cb_timeout(cb);
+      notify_client_cb_with_code(cb, OC_REQUEST_TIMEOUT);
```
Comment (Danielius1922) [inline api/oc_ri.c:1683]: I think this also shouldn't use  the `OC_REQUEST_TIMEOUT` code.

File: api/oc_ri.c [hunk 2]
```diff
@@ -1697,7 +1697,7 @@ oc_ri_free_client_cbs_by_endpoint(const oc_endpoint_t *endpoint)
     if (!cb->multicast && !cb->discovery && cb->ref_count == 0 &&
         oc_endpoint_compare(&cb->endpoint, endpoint) == 0) {
       cb->ref_count = 1;
-      notify_client_cb_timeout(cb);
+      notify_client_cb_with_code(cb, OC_CONNECTION_CLOSED);
```
Comment (Danielius1922) [inline api/oc_ri.c:1700]: Both `oc_ri_free_client_cbs_by_mid` and `oc_ri_free_client_cbs_by_endpoint` are public API, so they can be called whenever, not just when connection is closed. So I think `OC_CONNECTION_CLOSED` doesn't always correctly describe what's happening. I guess both functions cancel the callback, so maybe something like `OC_CANCELLED` ? 
Or maybe we should add a status code parameter, so they caller can determine which code should be received in the cancelled callback. Although that would be an API change.

Comment (jkralik) [reply api/oc_ri.c]: fixed

File: api/oc_ri.c (hunk 1 above)
Comment (jkralik) [reply api/oc_ri.c]: fixed

File: api/oc_ri.c [hunk 3]
```diff
@@ -1689,22 +1689,35 @@ oc_ri_free_client_cbs_by_mid(uint16_t mid)
 }
 
 void
-oc_ri_free_client_cbs_by_endpoint(const oc_endpoint_t *endpoint)
+oc_ri_free_client_cbs_by_mid(uint16_t mid)
+{
+  oc_ri_free_client_cbs_by_mid_v1(mid, OC_REQUEST_TIMEOUT);
```
Comment (Danielius1922) [inline api/oc_ri.c:1694]: Wouldn't `OC_CANCELLED` be a better default than a timeout? (I've already changed the value from `OC_STATUS_SERVICE_UNAVAILABLE` to `OC_REQUEST_TIMEOUT`, so I don't think changing the code value is a problem).

Comment (jkralik) [reply api/oc_ri.c]: Ok. I didn't know it. Fixed

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**